pysat/_instrument.py

```diff
@@ -392,12 +401,15 @@ def __init__(self, platform=None, name=None, tag=None, inst_id=None,
                 saved_keys.extend(fkwargs)
 
                 # If the function can't access this dict, use partial
+                # The functions that truly can't access the dict are load,
```

Comment:
Thanks for being specific :)

I was looking at the code to come up with the proper NOUN but given my changes a couple comments had drifted out of date. I put in a minimum set of comment changes to best reflect the intent of your comment, that was the plan at least.

Given how this pull has gone this shouldn't be surprising, but I think I few lines of code in that method should actually come out now that things are more settled. I'm going to poke at that (very limited) and report back. I'll leave the option of what to do with it to y'all once I have the solution.

Reply:
So the code in this area can be reduced to
```
        ...
        # Expected function keywords
        exp_keys = ['list_files', 'load', 'preprocess', 'download',
                    'list_remote_files', 'clean', 'init']
        for fkey in exp_keys:
            func_name = _kwargs_keys_to_func_name(fkey)
            func = getattr(self, func_name)

            # Get dict of supported keywords and values
            default_kwargs = _get_supported_keywords(func)

            # Check if kwargs are in list
            good_kwargs = [ckey for ckey in kwargs.keys()
                           if ckey in default_kwargs]

            # Store appropriate user supplied keywords for this function
            self.kwargs[fkey] = {gkey: kwargs[gkey] for gkey in good_kwargs}

            # Store all supported keywords for user edification
            self.kwargs_supported[fkey] = default_kwargs

            # Store keys to support check that all user supplied
            # keys are used.
            saved_keys.extend(default_kwargs.keys())

        # Test for user supplied keys that are not used
        ...
```

In the current pull attached functions (`_load_rtn`, `_list_files_rtn`, etc.) will always be partial functions. The changes above keep whatever function form is presented by the instrument support module. So for the testing instruments it can be a mixture of partial and regular functions. 

```
In [3]: test._load_rtn                                                                                                                                                                                                      
Out[3]: <function pysat.instruments.pysat_testing.load(fnames, tag=None, inst_id=None, sim_multi_file_right=False, sim_multi_file_left=False, root_date=None, malformed_index=False, num_samples=None, test_load_kwarg=None)>

In [4]: test._list_files_rtn                                                                                                                                                                                                
Out[4]: functools.partial(<function list_files at 0x7fb0d244fdc0>, test_dates={'': {'': datetime.datetime(2009, 1, 1, 0, 0), 'no_download': datetime.datetime(2009, 1, 1, 0, 0), 'non_strict': datetime.datetime(2009, 1, 1, 0, 0), 'user_password': datetime.datetime(2009, 1, 1, 0, 0), 'default_meta': datetime.datetime(2009, 1, 1, 0, 0)}})

```
 Per unit tests the end-level functionality is the same for both kinds of functions.

Technically there are options:
- [ ] Condense code, functions stay as they are given. This option is in code above as it is the minimum required. (unit tests pass)
- [ ] Keep the code as is in the pull now, everything is a partial function. 'Extra' lines do no harm. (unit tests pass)
- [ ] Condense code, make all functions partial functions (unit tests will pass)

How quickly things can spiral :) I can go for any of them really. Options 2 and 3 are more consistent than 1. That being said, the partialness of these functions isn't being directly used by pysat anymore with regards to keywords so no direct reason to make them all partial functions other than potential consistency.

Reply:
I should note that option 1 above is closest (?) to behavior that develop and the current actual RC branch have, mostly. Some instrument support functions are currently functions, others are partial. While some partial functions will always be partial, other functions remain how the instrument support module presents them.

So right now (develop-3) the `_*_rtn`s can be partial, or regular. The pull as is makes them all partial. Option 1 goes back to some are regular, some are partial.

Reply:
https://github.com/pysat/pysat/blob/4a2ed3a1f97ddbff5ae8c520ffe6d88b8ecf99a8/pysat/_instrument.py#L365

Reply:
I made #753 in an attempt to make this easier.

Reply:
Since this has been captured in a separate PR, I believe this can be resolved.